language/move-vm/runtime/src/loader.rs

```diff
@@ -82,14 +82,23 @@ impl ScriptCache {
         }
     }
 
-    fn get(&self, hash: &HashValue) -> Option<Arc<Function>> {
-        self.scripts.get(hash).map(|script| script.entry_point())
+    fn get(&self, hash: &HashValue) -> Option<(Arc<Function>, Vec<Type>)> {
```

Comment:
I really really hate the way I got the Loader to return parameter types for main and other functions. Tried a few other methods but they either didn't work due to the quirks of the Loader implementation or were massively more complicated than the current one.

I'll try a little bit harder and see if I can find a cleaner solution, but I think the fundamental issue here is the Loader's lack of modularity and well defined interfaces.  Let's also include "simplifying/remodeling the loader" as part of our plan to better understand the VM data flow.